### DISCUSSION TITLE
install to/bit from on board flash

### DISCUSSION THREAD

**Discussion by uglyfish333** [Joshua-Riek/ubuntu-rockchip#130] [Q&A] [answered]
Does anyone know/mind sharing the steps for installing/booting from the on board flash on the orange pi 5b? I see instructions for nvme or an SSD but not on board.

Edit: by on board I meM the emmc of the 5b, not the spi on the regular 5.

**Comment by Joshua-Riek** [comment, marked as answer]
It's a major pain to flash the emmc with an os. I personally would boot from an sd-card then download the image on your sd-card and flash it manually with the below commands.
```bash
xz -dc ubuntu-22.04.2-preinstalled-desktop-arm64-orangepi5b.img.xz | sudo dd of=/dev/mmcblk0 bs=4k
sync
```
However, there is a new video out that shows the process of installing the image using maskrom mode.
https://www.youtube.com/watch?v=5q_tytwmseg&t=925s

**Reply by uglyfish333** [reply]
Joshua,

Thank you for the super quick answer to my question, and for all of your work on this project.

I decided to follow your suggestion before watching the video, but I didn't even get t as far as opening up the terminal. After downloading the image in the browser I accidentally clicked to open the file instead of the location in the chromium downloads menu and up popped a prompt asking me if I wanted to restore the image, and one of the location options was a 128gb "SD Card Reader" which just so happens to be the size of the emmc. I selected the option, it wrote the image, and it installed. Turned off the pi, took out the SD card, booted up, went through the initial setup, and now I'm posting from me 5b, sans flash card. Couldn't be easier!

Thanks again!

**Reply by Joshua-Riek** [reply]
Awesome, i'm glad i could help! :)
